arex-instrumentation/httpclient/arex-httpclient-apache-v4/src/main/java/io/arex/inst/httpclient/apache/async/HttpAsyncMethodsInstrumentation.java

```diff
@@ -0,0 +1,47 @@
+package io.arex.inst.httpclient.apache.async;
+
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.arex.inst.extension.MethodInstrumentation;
+import io.arex.inst.extension.TypeInstrumentation;
+import io.arex.inst.httpclient.apache.common.ApacheHttpClientAdapter;
+import io.arex.inst.runtime.context.ContextManager;
+import java.util.Collections;
+import java.util.List;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.http.HttpRequest;
+
+/**
+ * Wrapped the http request for reading request body repeatedly.
+ * @since 2024/2/21
+ */
+public class HttpAsyncMethodsInstrumentation extends TypeInstrumentation {
+
+    @Override
+    protected ElementMatcher<TypeDescription> typeMatcher() {
+        return named("org.apache.http.nio.client.methods.HttpAsyncMethods");
+    }
+
+    @Override
+    public List<MethodInstrumentation> methodAdvices() {
+        return Collections.singletonList(new MethodInstrumentation(
```

Comment:
Is it better to change the modification method to org.apache.http.nio.protocol.BasicAsyncRequestProducer#BasicAsyncRequestProducer(org.apache.http.HttpHost, org.apache.http.HttpRequest)

Reply:
Good suggestion, I'll adjust it.